client/starwhale/api/_impl/model.py

```diff
@@ -31,16 +31,20 @@
 
 _TASK_ROOT_DIR = "/var/starwhale" if in_production() else "/tmp/starwhale"
 
-_p = lambda p, sub: Path(p) if p else Path(_TASK_ROOT_DIR) / sub
 _ptype = t.Union[str, None, Path]
+_p: t.Callable[[_ptype, str], Path] = (
+    lambda p, sub: Path(p) if p else Path(_TASK_ROOT_DIR) / sub
+)
```

Comment:
Do we really need the type hint for the lambda?  or just use `def` like 
```python
def _p(p: _ptype, sub: str) -> Path:
    return Path(p) if p else Path(_TASK_ROOT_DIR) / sub
``` 
The type follows the variable, making it clearer.
At the same time, I think the `_p` should have a more meaningful name.

Reply:
the original purpose: only need a shortcut, not too well-named alias function.
If not typing hint for lambda, we should add `#typing: ignore` for the lambda-used lines.